query/src/catalogs/backends/meta_backend.rs

```diff
@@ -70,7 +70,10 @@ impl MetaBackend {
         F: Send + Sync + 'static,
         T: Send + Sync + 'static,
     {
-        let cli = self.meta_api_provider.try_get_meta_client().await?;
+        let res: Result<_, std::convert::Infallible> =
+            self.meta_api_provider.try_get_meta_client().await;
+        let cli = res.unwrap();
```

Comment:
If we do the above return type modification, we do not need unwrap here.

Reply:
There are two different ways to make codes clean:
- To minimize the logic involved,
- or to follow a universal code style, e.g. a method that creates an instance always returns a `Result`.

In this case, either to minimize logic or to keep the return type of every `new_xxx()` method a `Result`. Both feel good to me.

It does not bother me. Then I'll take your word, sir:D

Reply:
Convinced by you :)